Removes lack of nuclear eguns nerfies

TobyThorne opened the pull request:
АУЕган с двух рук уверенно становится мемом, ибо практически раундстартом есть возможность получать бесконечный лазерный миниган.

- Режимы стрельбы изменены. Теперь, как и у обычного егана, есть стансферки и лазер. То же самое сделано для всех еганов, кроме требующих авторизацию.
- Штраф при стрельбе с одной руки увеличен с 1 до 2. Эта штука здоровенная (ITEM_SIZE_LARGE).
- Увеличено время перезарядки с 4 до 5 тиков. 
- Добавлен перегрев и связанное с ним излучение радиации и возможность взрыва.
В деталях:
У АУЕгана есть fail_counter.
При каждом выстреле fail_counter повышается на 2-4 единицы. С шансом в 10% он может подняться на 10-15 и заискриться.
Со временем fail_counter понижается и если он превышает 15, то излучается радиация.
Если fail_counter превышает 30, то при выстреле с вероятностью 67% оружие взорвётся.
Бездумное нажимание на курок (The advanced energy gun is not ready to fire again!) повышает fail_counter так же, как и нормальный выстрел.

Если fail_counter > 15 и оружие начинает фонить, то в чат (и в Examine) выведется:
"The advanced energy gun feels pleasantly warm."
Если fail_counter > 30 и следующий выстрел может привести в взрыву, то:
"The advanced energy gun feels burning hot!"

При осторожном применении АУЕган всё ещё крут, как яйца, ибо бесконечный, но использование его в качестве пулемёта уже не проканает.

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

Commit:
nerfing advanced energy gun

Voldirs commented:
>При каждом выстреле fail_counter повышается на 1-5 единиц. С шансом в 10% он может подняться на 10-15 и заискриться.

Так это даже полный заряд из него не настрелять без косяков. Не перебор ли?
Алсо трейторам закинь версию нормальную тогда. Я просто помню как раньше было, у напечатанных в рнд аеганов был параметр reliability, от которого считался шанс косяка, среди которых было облучение юзера и полное отключение перезарядки, то есть после второго аеган был полностью сломан, но если его пихнуть в деструктор, релайабилити для новых прототипов возрастал и это было очень красивое решение.

MrMagick2104 commented:
Имхо, стоит для пользователя добавить каких-то возможностей увидеть, что аеган остыл или ещё нагрет, желательно, при экзамайне писать:
Если криткаунтер выше 15, то добавлять что-то в стиле: The gun`s heatsink is a little bit red from the heat (либо про fan`s buzzing написать, в общем, система охлаждения на износ уже работает) and the reactor itself is faintly glowing green!
А если выше 30, то:
The gun`s reactor is violently shaking!
Примеры плохие, но идея, я думаю, понята.
Либо писать про спад фейлкаунтера в чат, без такого стрелять будет почти невозможно.
Хотелось бы увидеть, кроме того, увеличение радиационного фона при возрастании фейлкаунтера.

P.S. Вырабатывать электричество при помощи десяти аеганов, лежащих на тайле, а также коллекторов рядом? 🤔

dnivji commented:
Да, как-то жирно, есть шанс, что при первом же выстреле тебя ебанет радиацией. Нерф прям в духе бея. Нужно уверенно делать выстрелов 5 из ауегана, а вот шестой и дальше выстрел уже что бы был рискованным, а на 10+ с гарантией делал пакость (при непрерывной стрельбе).

TobyThorne commented:
> Имхо, стоит для пользователя добавить каких-то возможностей увидеть, что аеган остыл или ещё нагрет, желательно, при экзамайне писать:

Состояние реактора отражается на спрайте, на передней части пушки, где зелёная хреновина. Чем больше мигает белым - тем выше фейлкаунтер.
> Хотелось бы увидеть, кроме того, увеличение радиационного фона при возрастании фейлкаунтера.

Уже увеличивается.

Commit:
Update nuclear.dm

TobyThorne commented:
> Да, как-то жирно, есть шанс, что при первом же выстреле тебя ебанет радиацией. Нерф прям в духе бея. Нужно уверенно делать выстрелов 5 из ауегана, а вот шестой и дальше выстрел уже что бы был рискованным, а на 10+ с гарантией делал пакость (при непрерывной стрельбе).

Обновил. Снизил разброс при обычном выстреле с 1-5 до 2-4. И облучать будет, если fail_counter > 15, дабы при неудачном первом выстреле не ловить радиацию.

MrMagick2104 commented:
> Трейторам вообще его выпилить надо. Экспериментальная пушка из РнД в аплинке у триторов - это не смешное.

Да ладно. РнД сам по себе ~~не~~ смешной: изучаешь всякий мусор (например, магнитные катушки), которые уже есть В ОДНОМ ОТДЕЛЕ от РнД.
Плюс, можно оправдать, что украли где-то. У тех же ОБРщиков они тоже есть.

Moscowchief commented:
> Я просто помню как раньше было, у напечатанных в рнд аеганов был параметр reliability, от которого считался шанс косяка, среди которых было облучение юзера и полное отключение перезарядки, то есть после второго аеган был полностью сломан, но если его пихнуть в деструктор, релайабилити для новых прототипов возрастал и это было очень красивое решение.

Вот да. Правда в таком случае детям деструктора ничто не мешает прорашить их до рабочего состояния, но это все равно лучше, чем занерф в неюзабельное фонящее говнище.

MrMagick2104 commented:
> Правда в таком случае детям деструктора ничто не мешает прорашить их до рабочего состояния, но это все равно лучше, чем занерф в неюзабельное фонящее говнище. Нужна возможность довести говеный прототип до ума.

Там же прикол был в том, что вообще ВСЕ технологии страдают без прокачанного релайабилит. Т.е. тебе нужно рашить не только аеган, но ещё кучу всяких условных вортексов, БоХов, прочего. Приходится выбирать, короче, т.к. ресурсы не бесконечные.

TobyThorne commented:
> Вот да. Правда в таком случае детям деструктора ничто не мешает прорашить их до рабочего состояния

This. Если для получения нынешнего состояния достаточно несколько раз их поразбирать - ничего не изменится. Лишние две минуты дроча, а результат тот же.
>занерф в неюзабельное фонящее говнище. Нужна возможность довести говеный прототип до ума.

Ну, сейчас чутка подкрутил цифры, так что до неюзабельности очень далеко. Если стрелять аккуратно (или хотя бы подготовиться и взять, скажем, костюм радзащиты или антирадин), то всё ещё лучше стандартного егана. Ибо бесконечный заряд. 
Ну и да, он не взрывается сразу при достижении порогового fail_counter'а, а если попытаться выстрелить. Аккуратность - наше всё. Любители зажимать гашетку и поливать всё огнём из экспериментальной пушки с ядерным реактором сасат.

koronoshka commented:
А предупреждение прямо перед излучением радиации есть? Я бы эту облученную пушку перед выбросом в ебало врагам кидал и как рипер доставал новую

Moscowchief commented:
Ну короче говоря, да, если прикручивать reliability, то прикручивать сразу всему РнД, reliability только на аегане не имеет смысла.
В общем, все хорошо, но стоит еще прикрутить в экзамайн строчки по которым можно будет определить фонит ли аеаган или вообще вот-вот лопнет. На 15 появляется сообщение, что аеган чувствуется теплым, а на 30, что уже обжигающим.

MrMagick2104 commented:
> А предупреждение прямо перед излучением радиации есть? Я бы эту облученную пушку перед выбросом в ебало врагам кидал и как рипер доставал новую

Есть. Она искрится и меняет спрайтик, как выше сказано. 
> но стоит еще прикрутить в экзамайн строчки по которым можно будет определить фонит ли аеаган или вообще вот-вот лопнет. 

Спрайтик же меняется.

> Сейчас благодаря смешным прокачанным деталькам - бесконечные. Ставишь всякие фемто-манипуляторы и высираешь из одного куска металла тысячи пушечек.

Ну, как. Во всяком случае алмазы, даже с фемто-манипуляторами всякими, рассасываются крайне быстро, по ощущениям. 

P.S.
> Бездумное нажимание на курок (The advanced energy gun is not ready to fire again!) повышает fail_counter так же, как и нормальный выстрел.

Хм. А если впихнуть аеган в интегральную плату стрельбы и поставить тикер, который нажимает кнопку выстрела с задержкой не больше 1 десятой секунды, то что будет?
Или там КД самой платы не даст такие гранаты делать?

Commit:
Update nuclear.dm

TobyThorne commented:
Добавил сообщения после выстрела и в Examine.
Если fail_counter > 15 и оружие начинает фонить, то выводит
"The advanced energy gun feels pleasantly warm."
Если fail_counter > 30 и следующий выстрел может привести в взрыву, то
"The advanced energy gun feels burning hot!"

Moscowchief commented:
> Спрайтик же меняется.

Бля, в ОПе не было написано. А я уже представил смешные ситуации, когда, чтобы понять фонит ли лежащий посреди коридора аеган нужно рискнуть, подойти и потрогать его. Спрайт спилить бы. 
@TobyThorne Описывай в ОПе изменения поподробнее плз.

TapKo4eB commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -78,46 +78,85 @@
+	else if (fail_counter > 30)
+		to_chat(loc, SPAN("warning", "\The [src] feels burning hot!"))
+
+/obj/item/weapon/gun/energy/gun/nuclear/examine(mob/user)
+	..(user)
+	if(fail_counter > 15)
+		to_chat(user, "It feels pleasantly warm.")
+	else if (fail_counter > 30)
+		to_chat(user, "It feels burning hot!")
```
Мб проверку на проксимити? А то с расстояния можно учуять плеазант варм от пушки.

PlinHost replied:
This. Либо изменить сообщение, либо показывать только если моб близко.

TobyThorne replied:
Пофикшено.

TapKo4eB commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -78,46 +78,85 @@
 //override for failcheck behaviour
 /obj/item/weapon/gun/energy/gun/nuclear/Process()
-	if(fail_counter > 0)
+	if(fail_counter > 15)
```
Может быть стоит повысить до около 18 или даже 20? Ибо сообщение о теплоте лучше пусть будет предупреждать, а не сразу говорить о том что ты облучён.

PlinHost replied:
Предупреждало тебя `an experimental miniaturized reactor` в экзамине, надо было раньше думать.

TapKo4eB commented on code/modules/projectiles/guns/energy/nuclear.dm:
```diff
@@ -78,46 +78,85 @@
+
+/obj/item/weapon/gun/energy/gun/nuclear/Fire(atom/target, mob/living/user, clickparams, pointblank=0, reflex=0)
+	..()
+	if(fail_counter > 30)
```
Тут та же ситуация, человек даже не успеет почувствовать горячее состояние ауегана с красным спрайтом, когда он уже взлетит на воздух.

MrMagick2104 replied:
> Тут та же ситуация, человек даже не успеет почувствовать горячее состояние ауегана с критическим спрайтом, когда он уже взлетит на воздух.

Вот тут как раз нет, ибо тебе нужно выстрелить ещё 1 раз после того, как у тебя фейл каунтер выше 30 поднялся (судя по комментариями ОП-а)

PlinHost replied:
Раньше, чем ему выведется сообщение, никто не взлетит. А если не заметил это уже его проблемы.

Commit:
Update nuclear.dm

- Fixed overheat descriptions;
- Removed shock mode from the remaining energy guns (except for secure guns);
- Increased nuclear energy gun's recharge time (4 -> 5 ticks);
